python/dgl/contrib/dis_kvstore.py

```diff
@@ -473,6 +473,29 @@ def start(self):
                     data=res_tensor,
                     c_ptr=None)
                 _send_kv_msg(self._sender, back_msg, msg.rank)
+            # Init new data
+            elif msg.type == KVMsgType.NEW_DATA:
+                assert msg.rank == 0
+                data_str, target_name = msg.name.split('|')
+                data_name, data_type = self._deserialize_shared_tensor(data_str)
+                dtype = F.data_type_dict[data_type]
+                data_shape = F.asnumpy(msg.id).tolist()
```

Comment:
this looks weird. why is `data_shape` generated from `msg.id`?

Reply:
Changed.